From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Tue, 1 Nov 2011 16:45:55 +0100
Subject: [PATCH] Only generate an RPATH entry if LD_RUN_PATH is not empty, for
 cases where -rpath isn't specified.

This is http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=151024

This patch is still used in debian binutils 2.26-3

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 ld/emultempl/elf32.em | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/ld/emultempl/elf32.em b/ld/emultempl/elf32.em
index 0405d4fa7155..73c5447bf4da 100644
--- a/ld/emultempl/elf32.em
+++ b/ld/emultempl/elf32.em
@@ -1242,6 +1242,8 @@ fragment <<EOF
 	      && command_line.rpath == NULL)
 	    {
 	      lib_path = (const char *) getenv ("LD_RUN_PATH");
+	      if ((lib_path) && (strlen (lib_path) == 0))
+		  lib_path = NULL;
 	      if (gld${EMULATION_NAME}_search_needed (lib_path, &n,
 						      force))
 		break;
@@ -1523,6 +1525,8 @@ gld${EMULATION_NAME}_before_allocation (void)
   rpath = command_line.rpath;
   if (rpath == NULL)
     rpath = (const char *) getenv ("LD_RUN_PATH");
+  if ((rpath) && (strlen (rpath) == 0))
+    rpath = NULL;
 
   for (abfd = link_info.input_bfds; abfd; abfd = abfd->link.next)
     if (bfd_get_flavour (abfd) == bfd_target_elf_flavour)
